lib/rules/no-lonely-if.js

```diff
@@ -31,7 +34,30 @@ module.exports = {
                         parent.body.length === 1 && grandparent &&
                         grandparent.type === "IfStatement" &&
                         parent === grandparent.alternate) {
-                    context.report(node, "Unexpected if as the only statement in an else block.");
+                    context.report({
+                        node,
+                        message: "Unexpected if as the only statement in an else block.",
+                        fix(fixer) {
+                            const openingElseCurly = sourceCode.getFirstToken(parent);
+                            const closingElseCurly = sourceCode.getLastToken(parent);
+                            const elseKeyword = sourceCode.getTokenBefore(openingElseCurly);
+                            const closingIfParen = sourceCode.getTokenAfter(node.test);
+                            const ifKeyword = sourceCode.getTokenBefore(node.test, 1);
+                            const ifConsequentText = node.consequent.type === "BlockStatement" ? sourceCode.getText(node.consequent).slice(1, -1) : sourceCode.getText(node.consequent);
+                            const sourceText = sourceCode.getText();
+
+                            if (sourceText.slice(openingElseCurly.range[1], node.range[0]).trim() || sourceText.slice(node.range[1], closingElseCurly.range[0]).trim()) {
+
+                                // Don't fix if there are any non-whitespace characters interfering (e.g. comments)
+                                return null;
+                            }
+
+                            return fixer.replaceTextRange(
+                                [elseKeyword.range[1], closingElseCurly.range[0]],
+                                " " + sourceText.slice(ifKeyword.range[0], closingIfParen.range[1]) + sourceText.slice(elseKeyword.range[1], openingElseCurly.range[1]) + ifConsequentText
```

Comment:
Oops. I'll fix that.

Originally, I had tried to make the fixer a bit simpler; it would just remove the brackets after `else`:

```js
return fixer.replaceTextRange(
    [openingElseCurly.range[0], closingElseCurly.range[1]],
    " " + sourceCode.getText(ifStatementNode)
);
```

However, this caused some issues with ASI:

```js
if (a) {
} else {
  if (b) foo()
} bar()

// ↓ fix

if (a) {
} else if (b) foo() bar() // SyntaxError
```

However, I suppose we could avoid this by using similar logic to the autofixer from `curly`, and simply avoid fixing cases that would cause a SyntaxError.